src/pages/home/ReportScreen.js

```diff
@@ -276,111 +276,107 @@ class ReportScreen extends React.Component {
+                                        const skeletonViewContainerHeight = Math.round(event.nativeEvent.layout.height);
+
+                                        // Only set state when the height changes to avoid unnecessary renders
+                                        if (reportActionsListViewHeight === skeletonViewContainerHeight) return;
+
+                                        // The height can be 0 if the component unmounts - we are not interested in this value and want to know how much space it
+                                        // takes up so we can set the skeleton view container height.
+                                        if (skeletonViewContainerHeight === 0) {
+                                            return;
+                                        }
+                                        reportActionsListViewHeight = skeletonViewContainerHeight;
+                                        this.setState({skeletonViewContainerHeight});
+                                    }}
+                                >
+                                    {this.isReportReadyForDisplay() && !isLoadingInitialReportActions && !isLoading && (
+                                        <ReportActionsView
                                             reportActions={this.props.reportActions}
                                             report={this.props.report}
                                             isComposerFullSize={this.props.isComposerFullSize}
-                                            onSubmitComment={this.onSubmitComment}
-                                            policies={this.props.policies}
+                                            policy={policy}
                                         />
-                                    </>
-                                )}
+                                    )}
 
-                                {!this.isReportReadyForDisplay() && (
-                                    <ReportFooter
-                                        shouldDisableCompose
-                                        isOffline={this.props.network.isOffline}
-                                    />
-                                )}
-                            </View>
-                        </DragAndDropProvider>
-                    </FullPageNotFoundView>
-                </ScreenWrapper>
-            </ReportScreenContext.Provider>
+                                    {/* Note: The report should be allowed to mount even if the initial report actions are not loaded. If we prevent rendering the report while they are loading then
+                            we'll unnecessarily unmount the ReportActionsView which will clear the new marker lines initial state. */}
+                                    {(!this.isReportReadyForDisplay() || isLoadingInitialReportActions || isLoading) && (
```

Comment:
I think making this a ternary with the block above it will improve readability. It makes it easier to understand that the intention is that either one or the other should be displayed, never both.

Reply:
All these are part of linting (the component it is now wrapped into a new context provider), I have not changed or touched the internal logic. I can do it if you want, but it falls out of the scope of the performance improvements